src/prefect/settings.py

```diff
@@ -417,6 +417,24 @@ def warn_on_database_password_value_without_usage(values):
     prefetched. Defaults to `10`.""",
 )
 
+PREFECT_ASYNC_FETCH_STATE_RESULT = Setting(
+    bool,
+    default=False,
+    description=textwrap.dedent(
+        """
+        Determines whether `State.result()` fetches results automatically or not.
```

Comment:
See https://github.com/PrefectHQ/prefect/pull/7067 RE using attribute docstrings for the API docs

Reply:
Thanks!